csml_engine/src/db_connectors/dynamodb/bot.rs

```diff
@@ -0,0 +1,358 @@
+use crate::data::DynamoDbClient;
+use csml_interpreter::data::{csml_bot::DynamoBot, csml_flow::CsmlFlow,};
+use crate::db_connectors::dynamodb::{Bot, DynamoDbKey, Flows};
+use crate::{CsmlBot, EngineError};
+
+use std::collections::HashMap;
+use rusoto_dynamodb::*;
+
+use crate::db_connectors::dynamodb::utils::*;
+
+pub fn create_bot_version(
+    bot_id: String,
+    bot: String,
+    db: &mut DynamoDbClient,
+) -> Result<String, EngineError> {
+    let data: Bot = Bot::new(bot_id, bot);
+
+    let input = PutItemInput {
+        item: serde_dynamodb::to_hashmap(&data)?,
+        table_name: get_table_name()?,
+        ..Default::default()
+    };
+
+    let client = db.client.to_owned();
+    let future = client.put_item(input);
+    db.runtime.block_on(future)?;
+    Ok(data.id.to_owned())
+}
+
+fn create_flows_baches(mut flows: Vec<CsmlFlow>,) -> Vec<String> {
+    let mut flows_vec = vec!();
+    let mut max_range = flows.len();
+
+    while max_range > 0 {
+        let range = match max_range {
+            max_range if max_range > 10 => 9,
+            max_range => max_range
+        };
+
+        let tmp: Vec<CsmlFlow> = flows.drain(0..range).collect();
+
+        let serialize_flows = base64::encode(bincode::serialize(&tmp).unwrap());
+        flows_vec.push(serialize_flows);
+        max_range = max_range - range;
+
+    }
+
+    flows_vec
+}
+
+pub fn create_flows_batches(
+    bot_id: String,
+    id_bot: String,
+    flows: Vec<CsmlFlow>,
+    db: &mut DynamoDbClient,
+) -> Result<(), EngineError> {
+    let flows_vec = create_flows_baches(flows);
+
+    for flows in flows_vec {
+        let data: Flows = Flows::new(bot_id.clone(), id_bot.clone(), flows);
+        let input = PutItemInput {
+            item: serde_dynamodb::to_hashmap(&data)?,
+            table_name: get_table_name()?,
+            ..Default::default()
+        };
+        let client = db.client.to_owned();
+        let future = client.put_item(input);
```

Comment:
use the BatchWriteItem operation (https://rusoto.github.io/rusoto/rusoto_dynamodb/struct.BatchWriteItemInput.html - note: max 25 items per batch!) to call the DB only once and not at every loop.